src/main/scala/util/AsyncQueue.scala

```diff
@@ -78,17 +78,17 @@ class AsyncQueueSource[T <: Data](gen: T, params: AsyncQueueParams = AsyncQueueP
   val bits = params.bits
   val sink_ready = WireInit(true.B)
   val mem = Reg(Vec(params.depth, gen)) // This does NOT need to be reset at all.
-  val widx = GrayCounter(bits+1, io.enq.fire(), !sink_ready, "widx_bin")
+  val widx = withReset(reset.asAsyncReset)(GrayCounter(bits+1, io.enq.fire(), !sink_ready, "widx_bin"))
```

Comment:
if we have to do this .asAsyncReset anyway, is there any reason not to just keep the `AsyncResetReg` in `GrayCounter` (and not change this bit of code)?

Reply:
I thought one of the goals was to remove all the AsyncResetReg instantiations and replace them with standard chisel Reg driven with the appropriate type of reset.

Reply:
I think that is correct that was a goal, but when I look at this all I see is the same thing that `AsyncResetReg` is doing internally anyway. If we have to cast the reset (as in, we are explicitly putting down AsyncResetReg even if the reset coming in is not Async reset otherwise) then I dont know if this has an advantage over `AsyncResetReg`, which is no longer a black-box instantiator anyway.

I could be convinced otherwise, just gunshy of modifications to these primitives

Reply:
I think there is also a problem with reset type reassignment.  Trying this just now in an RF system, I get:
Exception in thread "main" firrtl.passes.PassExceptions: 
firrtl.passes.CheckTypes$InvalidConnect: : [module AsyncQueueSource]  Type mismatch in 'widx_bin.reset <= reset'.
  widx_bin.reset: UInt<1>
  reset: AsyncReset

Reply:
ah, i see, you can't actually can't use the `AsyncResetReg.apply()` without casing the reset yourself to `asAsyncReset` because:

https://github.com/chipsalliance/rocket-chip/blob/4f0cdea85c8a2b849fd582ccc8497892001d06b0/src/main/scala/util/AsyncResetReg.scala#L75

Ok, I am fine with this then

Reply:
but now that I look again at what `AsyncResetReg` is actually doing, yeah let's just use the native RegInit stuff :)